ansible/INSTALL.md

```diff
@@ -0,0 +1,165 @@
+### Configure opensds cluster variables:
+##### System environment:
+If you want to integrate OpenSDS with cloud platform (for example k8s), please modify `nbp_plugin_type` variable in `group_vars/common.yml`:
+```yaml
+nbp_plugin_type: standalone # standalone is the default integration way, but you can change it to 'csi', 'flexvolume'
+```
+
+##### LVM
+If `lvm` is chosen as storage backend, modify `group_vars/osdsdock.yml`:
+```yaml
+enabled_backend: lvm # Change it according to the chosen backend. Supported backends include 'lvm', 'ceph', and 'cinder'
+pv_devices: # Specify block devices and ensure them existed if you choose lvm
+  #- /dev/sdc
+  #- /dev/sdd
+vg_name: "specified_vg_name" # Specify a name for VG if choosing lvm
+```
+
+Modify ```group_vars/lvm/lvm.yaml```, change pool name to be the same as `vg_name` above:
+```yaml
+tgtBindIp: 127.0.0.1 # change tgtBindIp to your real host ip, run 'ifconfig' to check
+tgtConfDir: /etc/tgt/conf.d
+pool:
+  "vg001" # change pool name to be the same as vg_name
+```
+
+Besides, Change `tgtBindIp` variable in `group_vars/lvm/lvm.yaml` to your real host ip.
+
+##### Ceph
+If `ceph` is chosen as storage backend, modify `group_vars/osdsdock.yml`:
+```yaml
+enabled_backend: ceph # Change it according to the chosen backend. Supported backends include 'lvm', 'ceph', and 'cinder'.
+ceph_pools: # Specify pool name randomly if choosing ceph
+  - rbd
+  #- ssd
+  #- sas
+```
+
+Modify ```group_vars/ceph/ceph.yaml```, change pool name to be the same as `ceph_pool_name`. But if you enable multiple pools, please append the current pool format:
+```yaml
+"rbd" # change pool name to be the same as ceph pool
+```
+
+Configure two files under ```group_vars/ceph```: `all.yml` and `osds.yml`. Here is an example:
+
+```group_vars/ceph/all.yml```:
+```yml
+ceph_origin: repository
+ceph_repository: community
+ceph_stable_release: luminous # Choose luminous as default version
+public_network: "192.168.3.0/24" # Run 'ip -4 address' to check the ip address
+cluster_network: "{{ public_network }}"
+monitor_interface: eth1 # Change to the network interface on the target machine
+```
+```group_vars/ceph/osds.yml```:
+```yml
+devices: # For ceph devices, append ONE or MULTIPLE devices like the example below:
+    - '/dev/sda' # Ensure this device exists and available if ceph is chosen
+    - '/dev/sdb' # Ensure this device exists and available if ceph is chosen
+osd_scenario: collocated
+```
+
+### Check if the hosts can be reached
+```bash
+sudo ansible all -m ping -i local.hosts
+```
+
+### Run opensds-ansible playbook to start deploy
+```bash
+sudo ansible-playbook site.yml -i local.hosts
+```
+
+## Test it
+### Configure opensds CLI tool
+```bash
+sudo cp /opt/opensds-linux-amd64/bin/osdsctl /usr/local/bin
+export OPENSDS_ENDPOINT=http://127.0.0.1:50040
+export OPENSDS_AUTH_STRATEGY=noauth
+osdsctl pool list # Check if the pool resource is available
+```
+
+### Check if the csi plugin pod running
+```bash
+kubectl get po
```

Comment:
Add some text here to say this is if you have set nbp_plugin_type to csi.

Reply:
kubectl get pod

Reply:
Ok, BTW the `get po` is another style of kubectl command, as well as `svc` and `ns`. If you think user would be confused , I will change it.